common/Unit.cpp

```diff
@@ -535,7 +541,7 @@ void UnitBase::endTest(const std::string& reason)
     _socketPoll->joinThread();
 
     // tell the timeout thread that the work has finished
-    TimeoutThreadMutex.unlock();
```

Comment:
Manually unlocking here was wrong because 1) we didn't take the lock and 2) the Mutex might already be unlocked.

Both cases are invalid.